discovery/dns/dns.go

```diff
@@ -317,19 +317,22 @@ func askServerForName(name string, queryType uint16, client *dns.Client, servAdd
 	}
 
 	response, _, err := client.Exchange(msg, servAddr)
-	if err == dns.ErrTruncated {
+	if err != nil {
+		if err == dns.ErrId {
```

Comment:
Do we really need to special case this?

Reply:
if we don't, the response may be nil and inappropriately dereferenced; this is one of many solutions. Happy to take whatever the maintainers agree on.

Reply:
I'm ok having only:

```go
        response, _, err := client.Exchange(msg, servAddr)
	if err != nil {
		return nil, err
	}
```

I suspect that the previous `if msg.Id != response.Id {...}` block was there for historical reasons (eg the dns library didn't return `dns.ErrId` at that time).

Reply:
Not an issue, but I'd like to request a quorum of agreement before I make any more changes to this. Again, happy to, just optimizing the process a bit I hope.

Reply:
@simonpasquier That's also my suspicion.

Reply:
(Chipping in as a non-maintainer): I completely understand the @brian-brazil 's comment and I think it should be fine even if we stop special casing Id missing error. To explain my review comments, I had suggested we special case that error only because it was already being done before and this PR was essentially made for upgrading the `miekg/dns` dependency. (Moreover, the new `ErrId` wraps a string that says `Id mismatch` rather than show the two mistmatching Ids).

In any case, I think the changes should be fine either way. Can we request consensus among the maintainers so that @erikh is able to complete this PR?